perf-tool/include/server.hpp

```diff
@@ -61,7 +61,7 @@ class Server
      */
 protected:
 public:
-    Server(int portNo, std::string commandFileName = "", std::string logFileName = "logs.txt");
+    Server(int portNo, const std::string &commandFileName = "", const std::string &logFileName = "logs.txt");
```

Comment:
I'm not sure what the main motivation behind this is, but if it's related to minimizing the cost of copy construction, in C++11 I believe most implementations have the Small/Short String Optimization which makes copy construction for small strings cheaper than copying the const ref + the indirection for field accesses. That said, looks like xLC provides `ibm::sso_string` instead of implementing it in `std::string` so 🤷🏾‍♂️ .